app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -107,66 +115,55 @@
       });
     };
 
-    var updateEnvFromEntries = function(entries) {
-      ctrl.envFromEntries = entries || [];
+    var updateEnvFromEntries = function() {
+      var optionsByName = {};
+      var entriesMap = {};
+
+      ctrl.envFromEntries = ctrl.entries || [];
 
       if(!ctrl.envFromEntries.length) {
         addEntry(ctrl.envFromEntries);
       }
+      _.each(ctrl.envFromSelectorOptions, function(option) {
+        optionsByName[option.metadata.name] = option;
+      });
```

Comment:
```js
var secretsByName = {};
var configMapsByName = {};
_.each(ctrl.envFromSelectorOptions, function(option) {
  switch(option.kind) {
    case 'Secret':
      secretsByName[option.metadata.name] = option;
      break;
    // ...
  }
});
```

Reply:
Select with `.has-warning` and `.help-block`
<img width="615" alt="screen shot 2017-10-20 at 3 37 30 pm" src="https://user-images.githubusercontent.com/1874151/31839100-a43f74ba-b5ad-11e7-992e-63ad6fdfffb0.png">